### ISSUE TITLE
gitsigns spacing and colour

### ISSUE THREAD

**Issue by sho-87** [dstein64/nvim-scrollview#129]
ive noticed a couple of problems with gitsigns re: spacing and colour

### Spacing

![image](https://github.com/dstein64/nvim-scrollview/assets/5199715/220b8b9e-f5be-4b7e-9597-76039341773f)

if you look at gitsigns on the left of my screen you'll see changes to 3 consecutive rows.

in scrollview on the right there is additional space between each sign, as well as the mark

looks like this is because the #lines is less than the height of the window, so the scrollview needs to stretch the placement of things to cover the entire height. this can be confirmed by just changing the height of the nvim window:

![image](https://github.com/dstein64/nvim-scrollview/assets/5199715/db981009-10fc-465b-8f89-a36dae1663f6)

### Colour

the colour for added lines is incorrect. it shows up as yellow but, as you can see from the left of the image above, gitsigns has it configured as green

**Comment by dstein64** [comment]
The spacing issue is addressed as of PR #130. When `scrollview_always_show` is used and a sign for `gitsigns` would correspond to multiple rows of the scrollview column, all those rows will be used.

@sho-87, can you provide steps for reproducing the other issue? Something like:

```sh
$ git clone <REPO>
$ git checkout <COMMIT>
$ nvim
# plus a list of edits to the file to reproduce the issue
```

**Comment by sho-87** [comment]
@dstein64 

to reproduce, here is a fresh config that bootstraps lazy and grabs scrollview and gitsigns at specific commits:

init.lua:
``` lua
local lazypath = vim.fn.stdpath 'data' .. '/lazy/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
  local lazyrepo = 'https://github.com/folke/lazy.nvim.git'
  vim.fn.system { 'git', 'clone', '--filter=blob:none', '--branch=stable', lazyrepo, lazypath }
end ---@diagnostic disable-next-line: undefined-field
vim.opt.rtp:prepend(lazypath)

require('lazy').setup({
  {
    "lewis6991/gitsigns.nvim",
    commit = "078041e",
    config = function(_, opt)
      require('gitsigns').setup {
        signcolumn = true,
    }
    end
  },
  {
    "dstein64/nvim-scrollview",
    commit = "1d80f72",
    config = function(_, opts)
      require("scrollview").setup(opts)
      require("scrollview.contrib.gitsigns").setup(opts)
    end,
    opts = {
      always_show = true,
      column=40,
    }
  },
})
```

1. `git init` in a directory, create a new file, and commit it. I just used some random text like:
```
jkh
jklsdfh
sdf
sdf
sdf
sdf
sdf
sdf
sdf
sdfdsf
asdasdasd
isdfsdf
```
2. nvim using the config above
3. add a new line to the end of the file and you'll see this, which is the correct colour for an addition:
![image](https://github.com/dstein64/nvim-scrollview/assets/5199715/97ceea9a-9596-48e9-994c-f99e95d85c90)
4. go to the line _above_ and change it, and this is where it goes wrong:
![image](https://github.com/dstein64/nvim-scrollview/assets/5199715/94ee3627-0efa-428f-8642-227173a0df55)

the marker is still there for the line addition, but the colour that used to be the correct blue has now turned into purple

**Comment by dstein64** [comment]
The second issue is fixed in 5a7eb7e6c1b921761615b57a6140d73b1cc2b034. A `gitsigns` change hunk could be comprised of a change and an addition. The existing code was assuming it was only a change, with no addition.

@sho-87, thanks for reporting the issue and providing the detailed reproduction steps. If you're still encountering a problem, please let me know with a new Issue or a comment here.

**Comment by sho-87** [comment]
awesome! thanks for your speed
